File: App/Clipboard.cpp
#include <pch.h>
#include "Clipboard.h"

namespace {
    std::unique_ptr<Clipboard> clipboard;
}

Clipboard::Clipboard()
{
}

Clipboard::~Clipboard()
{
}

Clipboard* Clipboard::get()
{
    if(!clipboard) {
        clipboard = std::make_unique<Clipboard>();
	}
    return clipboard.get();
}

void Clipboard::readText(const rapidjson::Value& params, JsonResult* result)
{
}

void Clipboard::writeText(const rapidjson::Value& params, JsonResult* result)
{
}

void Clipboard::readImage(const rapidjson::Value& params, JsonResult* result)
{
}

void Clipboard::writeImage(const rapidjson::Value& params, JsonResult* result)
{
}

void Clipboard::readFile(const rapidjson::Value& params, JsonResult* result)
{
}

void Clipboard::writeFile(const rapidjson::Value& params, JsonResult* result)
{
}

void Clipboard::readHtml(const rapidjson::Value& params, JsonResult* result)
{
}

void Clipboard::writeHtml(const rapidjson::Value& params, JsonResult* result)
{
}

void Clipboard::readRtf(const rapidjson::Value& params, JsonResult* result)
{
}

void Clipboard::writeRtf(const rapidjson::Value& params, JsonResult* result)
{
}

void Clipboard::readBookmark(const rapidjson::Value& params, JsonResult* result)
{
}

void Clipboard::writeBookmark(const rapidjson::Value& params, JsonResult* result)
{
}

void Clipboard::clear(const rapidjson::Value& params, JsonResult* result)
{
}

File: Script/Clipboard.ts
import { Eventer } from "./Eventer";
export class Clipboard extends Eventer {
  getDataType() {
    return this.execute("getDataType");
  }
  readText() {
    return this.execute("readText");
  }
  writeText(text: string) {
    return this.execute("writeText", text);
  }
  readHtml() {
    return this.execute("readHtml");
  }
  writeHtml(html: string) {
    return this.execute("writeHtml", html);
  }
  readRtf() {
    return this.execute("readRtf");
  }
  writeRtf(rtf: string) {
    return this.execute("writeRtf", rtf);
  }
  readImg() {
    return this.execute("readImg");
  }
  writeImg(rtf: string) {
    return this.execute("writeImg", rtf);
  }
  getFile() {
    return this.execute("getFile");
  }
  addFile(...paths) {
    return this.execute("addFile", ...paths);
  }
  clear() {
    return this.execute("clear");
  }
  private execute(methodName: string, ...params: any[]) {
    return this.call({
      className: "clipboard",
      winId: horse.win.id,
      methodName,
      params,
    });
  }
}
